pandas/_libs/src/datetime/np_datetime_strings.c

```diff
@@ -444,16 +272,6 @@ int parse_iso_8601_datetime(char *str, int len, PANDAS_DATETIMEUNIT unit,
             *out_special = 1;
         }
```

Comment:
do we need an assertion here that unit is FR_ns?

Reply:
To double-check, what line are you referring to?  The answer is "no" regardless, but the reason why may vary.

Reply:
parse_iso_8601_datetimetime takes a unit arg, is this vaidated that it is only ever FR_ns

Reply:
OK.  The answer is "no" and the reason is because 1) the `unit` arg is no longer used once we've gotten rid of the `casting` arg (so it'll be removed in a follow-up) and 2) `parse_iso_8601_datetime` is only ever called from src/datetime.pxd and in that one case it is with PANDAS_FR_ns.

Reply:
ok that's what I figured, ok if you are removing then this is ok.